codegen/frameCodegen.ts

```diff
@@ -0,0 +1,199 @@
+import { hex } from "../crypto/mod.ts"
+import * as $ from "../deps/scale.ts"
+import { FrameMetadata } from "../frame_metadata/mod.ts"
+import { CodecCodegen } from "./CodecCodegen.ts"
+import { TypeCodegen } from "./TypeCodegen.ts"
+
+export function frameCodegen(
+  metadata: FrameMetadata,
+  chainIdent: string,
+  files: Map<string, string>,
+) {
+  const codecCodegen = new CodecCodegen()
+  codecCodegen.visit(metadata)
+  codecCodegen.write(files)
+
+  const typeCodegen = new TypeCodegen(codecCodegen, metadata.types)
+  typeCodegen.write(files)
+
+  files.set("mod.d.ts", MOD)
+  files.set("mod.js", MOD)
+
+  files.set(
+    "metadata.d.ts",
+    `
+      ${IMPORTS_COMMON}
+      export const metadata: metadata
+      export type metadata = ${typeCodegen.print(metadata)}
```

Comment:
```suggestion
      export const metadata: ${typeCodegen.print(metadata)}
```

Reply:
I believe there's only one place where we do this, and `typeof metadata` seems much more semantic than having a lower-cased `metadata` type.